internal/provider/azure/environ.go

```diff
@@ -233,19 +233,64 @@ func (env *azureEnviron) initEnviron(ctx context.Context) error {
 // PrepareForBootstrap is part of the Environ interface.
 func (env *azureEnviron) PrepareForBootstrap(ctx environs.BootstrapContext, _ string) error {
 	if ctx.ShouldVerifyCredentials() {
-		if err := verifyCredentials(env, ctx); err != nil {
+		if err := verifyCredentials(ctx, env); err != nil {
 			return errors.Trace(err)
 		}
 	}
 	return nil
 }
 
-// Create is part of the Environ interface.
-func (env *azureEnviron) Create(ctx envcontext.ProviderCallContext, args environs.CreateParams) error {
-	if err := verifyCredentials(env, ctx); err != nil {
+// ValidateModelCreation is part of the [environs.ModelResources] interface.
+func (env *azureEnviron) ValidateModelCreation(ctx context.Context) error {
+	if env.config.resourceGroupName == "" {
+		return nil
+	}
+	return env.validateResourceGroup(ctx, env.resourceGroup)
+}
+
+func (env *azureEnviron) validateResourceGroup(ctx context.Context, resourceGroupName string) error {
+	resourceGroups, err := env.resourceGroupsClient()
+	if err != nil {
 		return errors.Trace(err)
 	}
-	return errors.Trace(env.initResourceGroup(ctx, args.ControllerUUID, env.config.resourceGroupName != "", false))
+	logger.Debugf(ctx, "validating existing resource group %q for model %q", resourceGroupName, env.modelName)
+	g, err := resourceGroups.Get(ctx, resourceGroupName, nil)
+	if err != nil {
+		return env.HandleCredentialError(ctx, errors.Annotatef(err, "checking resource group %q", resourceGroupName))
+	}
+	if region := toValue(g.Location); region != env.location {
+		return errors.Errorf("cannot use resource group in region %q when operating in region %q", region, env.location)
+	}
+	modelTag := names.NewModelTag(env.config.Config.UUID())
```

Comment:
nit: up to you if you want to fix this as another drive by. Model tag isn't needed here reading the code. It is the uuid this function uses and needs.

Swapping them out makes another sacrifice to the tag overlords.

Reply:
yes indeed. been fixing as i go, missed this one